application/src/test/java/org/thingsboard/server/system/BaseHttpDeviceApiTest.java

```diff
@@ -60,6 +67,22 @@ public void before() throws Exception {
                 doGet("/api/device/" + device.getId().getId().toString() + "/credentials", DeviceCredentials.class);
     }
 
+    @Test
+    public void testRateLimits_wrongCredentialsMaxOut() throws Exception {
+        doGetAsync("/api/v1/" + deviceCredentials.getCredentialsId() + "/attributes?clientKeys=keyA,keyB,keyC")
+                .andExpect(status().isOk());
+
+        for (int i = 0; i < maxWrongCredentialsPerIp; i++){
+            doGetAsync("/api/v1/" + "WRONG_TOKEN" + "/attributes?clientKeys=keyA,keyB,keyC")
```

Comment:
TransportService.process should be invoked `maxWrongCredentialsPerIp` times
plz check it